### PR TITLE
Feature/readme overview

### PR DESCRIPTION
**Pull request by amichnia**
Added:
 + animated overview gif images (needs update for verify)
 + descriptions for global Verify and Given methods
Changed:
 * updated readme with overview, examples, description of whole process

### CODE REVIEW

**Comment by amichnia** [comment]
[Prepare basic readme documentation](https://trello.com/c/yGRtj7ue/4-prepare-basic-readme-documentation)

File: README.md [hunk 1]
```diff
@@ -1,165 +1,268 @@
...
++ **templates**: path to Mocky sourcery templates, in Pods directory
++ **output**: place where `Mock.generated.swift` will be placed
++ **testable**: specify imports for Mock.generated, that should be marked as `@testable` (usually tested app module)
++ **import**: all additional imports, external libraries etc. to be placed in Mock.generated
+
+## Generate mocks:
+
+1. **manually**: by triggering:
+
+  `Pods/Sourcery/bin/sourcery --config <your config yml path> --disableCache`
```
Comment (CurlyHeir) [inline README.md]: we shouldn't disable cache for mocks generation, it was only for template development.

File: README.md [hunk 2]
```diff
@@ -1,165 +1,268 @@
...
+
+**Don't forget** to add `Mock.generated.swift` to your test target :)
+
+> **Please Note!**
+> Most convenient way is to put generating script in some kind of script - liek Rakefile below.
+> Mock generation is triggered by `rake mock`
+> ```ruby
+> # Rakefile
+> task :mock do
+>   sh "Pods/Sourcery/bin/sourcery --config mocky.yml --disableCache"
```
Comment (CurlyHeir) [inline README.md]: same here, we don't need `--disableCache` flag

File: README.md [hunk 3]
```diff
@@ -1,165 +1,268 @@
...
 
 - [x] stubbibg protocols in elegant way
-- [x] template for generating mocks 
+- [x] template for generating mocks
 - [x] example project
-- [x] stubbing protocols with variables ( for now, IUO and Optional types are supported) 
-- [ ] method signature generation without name conflicts 
-- [ ] cover 95% of framework codebase with unit tests 
-- [ ] refactor .swifttemplate to .stencil template, to improve performance of generation
+- [x] stubbing protocols with variables ( for now, IUO and Optional types are supported)
```
Comment (CurlyHeir) [inline README.md]: don't we support all types of variables now?

File: README.md [hunk 4]
```diff
@@ -1,165 +1,268 @@
...
-        return all.last?.returns
-    }
-// sourcery:end
+
++ **sources**: all directories you want to scan for protocols/files marked to be auto mocked, or inline mocked
++ **templates**: path to Mocky sourcery templates, in Pods directory
++ **output**: place where `Mock.generated.swift` will be placed
++ **testable**: specify imports for Mock.generated, that should be marked as `@testable` (usually tested app module)
++ **import**: all additional imports, external libraries etc. to be placed in Mock.generated
+
```
Comment (CurlyHeir) [inline README.md]: i'd be nice to skip scanning files that ends with `.generated.swift`, files generated by `R.swift` are quite big, please add ticket for that.

Comment (amichnia) [reply README.md]: Good idea, adding issue and ticket

**Pull request merged**